test/functional/aws_test.go

```diff
@@ -0,0 +1,242 @@
+			StorageEncrypted:   aws.Bool(true),
+		},
+		{
+			DBName:             aws.String(CRResourceName),
+			DeletionProtection: aws.Bool(true),
+			MultiAZ:            aws.Bool(true),
+			StorageEncrypted:   aws.Bool(true),
+		},
+		{
+			DBName:             aws.String(CRResourceName),
+			DeletionProtection: aws.Bool(true),
+			MultiAZ:            aws.Bool(true),
+			StorageEncrypted:   aws.Bool(true),
+		},
+		//TODO implement Fuse here when it's ready
+		{
+			DBName:             aws.String(ClusterSSOResourceName),
+			DeletionProtection: aws.Bool(true),
+			MultiAZ:            aws.Bool(true),
+			StorageEncrypted:   aws.Bool(true),
+		},
+		{
+			DBName:             aws.String(ApplicationSSOResourceName),
+			DeletionProtection: aws.Bool(true),
+			MultiAZ:            aws.Bool(true),
+			StorageEncrypted:   aws.Bool(true),
+		},
+	}
+	expectedElasticacheResources = []*elasticache.ReplicationGroup{
+		{
+			AtRestEncryptionEnabled:  aws.Bool(true),
+			TransitEncryptionEnabled: aws.Bool(true),
+		},
+	}
+	expectedS3Resources = []*s3.PublicAccessBlockConfiguration{
+		{
+			BlockPublicAcls:       aws.Bool(true),
+			BlockPublicPolicy:     aws.Bool(true),
+			IgnorePublicAcls:      aws.Bool(true),
+			RestrictPublicBuckets: aws.Bool(true),
+		},
+	}
+)
+
+func AWSResourcesExistTest(t *testing.T, ctx *TestingContext) {
+	err := setAWSCredEnvar(ctx.framework)
+	if err != nil {
+		t.Errorf("failed to set AWS cred envars : %w", err)
+	}
+	clusterID, err := common.GetClusterID(ctx.framework)
+	if err != nil {
+		t.Errorf("failed to get clusterID : %w", err)
+	}
+	instances, instanceErr := getRDSResources(*newAWSClients())
+	elasticacheResourcesMap, elasticacheErr := getElasticacheResources(*newAWSClients(), clusterID)
+	s3ResourcesMap, s3Err := getS3Resources(*newAWSClients(), clusterID)
+	if instanceErr != nil || elasticacheErr != nil || s3Err != nil {
+		t.Errorf("error getting one or more aws resource")
+	}
+	resourceMatchesExpectedState := make(map[string]bool)
+	//Check RDS
+	for _, expectedInstance := range expectedRDSResources {
+		for _, returnedInstance := range instances {
+			if expectedInstance.DBName != returnedInstance.DBName {
+				continue
+			}
+			if !common.EqualResourceBool(
+				[]bool{aws.BoolValue(expectedInstance.DeletionProtection), aws.BoolValue(expectedInstance.MultiAZ), aws.BoolValue(expectedInstance.StorageEncrypted)},
+				[]bool{aws.BoolValue(returnedInstance.DeletionProtection), aws.BoolValue(returnedInstance.MultiAZ), aws.BoolValue(returnedInstance.StorageEncrypted)}) {
+				t.Logf("RDS resources are not in the expected state")
+				resourceMatchesExpectedState[aws.StringValue(expectedInstance.DBName)] = false
+			}
+			resourceMatchesExpectedState[aws.StringValue(expectedInstance.DBName)] = true
+
+		}
+	}
+	//Check Elasticache
+	for _, expectedReplicationGroup := range expectedElasticacheResources {
+		for arn, replicationgroups := range elasticacheResourcesMap {
+			if strings.Contains(arn, ThreeScaleNameElement) {
```

Comment:
If I understand it correctly, this will make sure that if there is 3scale resource, it has correct configuration, but it won't fail if the resource is not there. Should we also check that the resources are there? (Same for other resource types)

Reply:
The code here is a bit outdated, but essentially there will be a final check that will ensure `resourceMatchesExpectedState` returns all true, else it will return what fields are missing or misconfigured, the results for Elasticache and s3 will be added into this map aswell :+1: